eortologio@danchris.github.io/helpers.js

```diff
@@ -1,69 +1,70 @@
-const GLib = imports.gi.GLib;
-const Me = imports.misc.extensionUtils.getCurrentExtension();
+import GLib from 'gi://GLib';
+import Gio from 'gi://Gio';
 
-function getNameDays(currentDatetime){
+export function getNameDays(currentDateTime){
 
     let nameDays = [];
-    let date = getCurrentDate(currentDatetime);
-    let [easterDay, easterMonth, easterYear] = calcOrthEaster(currentDatetime.get_year());
-    
-    return nameDays.concat(getRecurringNameDays(date), getRelativeToEasterNameDays(easterDay, easterMonth, easterYear, currentDatetime));
-    
+    let date = getCurrentDate(currentDateTime);
+    let [easterDay, easterMonth, easterYear] = calcOrthEaster(currentDateTime.get_year());
     
+    return nameDays.concat(getRecurringNameDays(date), getRelativeToEasterNameDays(easterDay, easterMonth, easterYear, currentDateTime));
+   
 }
 
-function getRecurringNameDays(date){
+export function getRecurringNameDays(date, subdir){
+    const file = Gio.File.new_for_uri(import.meta.url).get_parent().get_child('recurring_namedays.json'); 
+    const [, contents] = file.load_contents(null);
 
-    let filePath = Me.dir.get_child('recurring_namedays.json').get_path();
-   
     let recurringNameDays = [];
-    if (filePath){
-        let namedaysFile = GLib.file_get_contents(filePath)[1];
-        let jsonData = JSON.parse(namedaysFile);
-        jsonData.data.forEach(function(element){
-            if (element.date === date) {
-                recurringNameDays = recurringNameDays.concat(element.names);
-            }
-        });
-       GLib.free(namedaysFile);
-    }
-
-    
+    if (contents) {
+        const namedaysFile = contents.toString();
+        const jsonData = JSON.parse(namedaysFile);
+    jsonData.data.forEach(function(element) {
+        if (element.date === date) {
+            recurringNameDays = recurringNameDays.concat(element.names);
+        }
+    });
+}
     return recurringNameDays;
 }
 
-function getRelativeToEasterNameDays(easterDay, easterMonth, easterYear, currentDatetime){
+export function getRelativeToEasterNameDays(easterDay, easterMonth, easterYear, currentDateTime, subdir){
     
     let easterDateTime = GLib.DateTime.new(GLib.TimeZone.new_local(),easterYear, easterMonth, easterDay, 0,0,0);
-   
-    let filePath = Me.dir.get_child('relative_to_easter.json').get_path();
-  
+    const file = Gio.File.new_for_uri(import.meta.url).get_parent().get_child('relative_to_easter.json'); 
```

Comment:
Here you create a pointer. Please, consider to free the memory properly before return from the function.

Reply:
Understood I will work on this

Reply:
per andyholmes he said we are not supposed to free the memory properly before the return from the function.

Reply:
We are just assigning a value to a variable, when the scope ends it will be collected (i.e. freed)